packages/server/routes/api/images.js

```diff
@@ -0,0 +1,196 @@
+const express = require("express");
+const router = express.Router();
+const { validate } = require("uuid");
+
+const AuthController = require("../../controllers/auth");
+const ImagesController = require("../../controllers/images");
+
+const VALID_IMAGE_TYPES = [
+  "user_icon",
+  "guild_icon",
+  "guild_banner",
+  "event_banner",
+];
+
+router.put(
+  "/:type/:UUID",
+  AuthController.authenticate,
+  async (request, response) => {
+    const imageType = request.params.type;
+    const imageUUID = request.params.UUID;
+    const imageData = request.body.imageData;
+    
+    if (!VALID_IMAGE_TYPES.includes(imageType)) {
+      response.status(400).json({
+        status: "fail",
+        data: {
+          imageType: `Image type '${imageType}' is invalid.`,
+        },
+      });
+      return;
+    }
+    if (!validate(imageUUID)) {
+      response.status(400).json({
+        status: "fail",
+        data: {
+          imageUUID: `Image UUID '${imageUUID}' is invalid.`,
+        },
+      });
+      return;
+    }
+    if (!imageData) {
+      response.status(400).json({
+        status: "fail",
+        data: {
+          imageData: "Request body missing value 'imageData'.",
+        },
+      });
+      return;
+    }
```

Comment:
I didn't put this in the RFC so that's my bad, but we should also check to make sure the Base64 string we receive is actually encoded JPEG data. This goes for the imageData validation checks in the other route handlers, too. I added details in the RFC on how to implement this.